### PR TITLE
Ensure ensembles reconfigure before nodes exit

### PR DESCRIPTION
**Pull request by andrewjstone**
See specific commit messages for more details

### CODE REVIEW

**Comment by andrewjstone** [comment]
Part of solution to #572

File: src/riak_core_claimant.erl [hunk 1]
```diff
@@ -763,19 +763,22 @@ bootstrap_root_ensemble(Ring) ->
...
-    Failed = [Result || Result <- L,
-                        Result =/= ok],
-    (Failed =:= []) orelse reset_ring_id(self()),
+    _ = maybe_reset_ring_id(L),
 
     RootNodes = [Node || {_, Node} <- RootMembers],
     RootAdd = Members -- RootNodes,
     RootDel = RootNodes -- Members,
+
+    Res = [riak_ensemble_manager:remove(node(), N) || N <- RootDel,
```
Comment (lordnull) [inline src/riak_core_claimant.erl:778]: Dialyzer com pains about this not be an existent function. Looking at the riak_ensemble_manager code, this doesn't seem to exist. Is there something I'm missing, or was this functionality changed at some point?

Comment (reiddraper) [inline src/riak_core_claimant.erl:778]: cc @jtuple @andrewjstone

Comment (andrewjstone) [inline src/riak_core_claimant.erl:778]: This fix is spread over multiple repos. That function is added in this
commit to riak_ensemble (part of basho/riak_ensemble#19) still waiting for review:
https://github.com/basho/riak_ensemble/commit/fbc02b12af10936787fdff74c6c13ada1b24e45c

Comment (andrewjstone) [inline src/riak_core_claimant.erl:778]: cc @lordnull @reiddraper ^

Comment (andrewjstone) [inline src/riak_core_claimant.erl:778]: Here's the overarching issue: https://github.com/basho/riak_core/issues/572

Comment (lordnull) [inline src/riak_core_claimant.erl:778]: Dialyzer complains about different things with the associated riak_ensemble branch. The new complaints appear to be ignored warnings that haven't had the line number adjusted yet.

**Comment by jtuple** [comment]
+1

Individual pull-request looks good. This is part of a larger set of pull-requests. Entire set of changes was reviewed/tested as part of basho/riak_test#593

FYI, looks like I forgot to assign myself to this review when I started reviewing these PRs a few weeks ago. Sorry for the duplicate effort @lordnull, but thanks for the extra eyes. Finally got back to this review and completed it, so going to go ahead and let it get merged in.

**Comment by andrewjstone** [comment]
:+1:  69cb5c4

**Comment by andrewjstone** [comment]
@borshop merge

**Pull request merged**